sdks/go/pkg/beam/runners/direct/direct.go

```diff
@@ -72,7 +72,9 @@ func Execute(ctx context.Context, p *beam.Pipeline) error {
 	if err = plan.Down(ctx); err != nil {
 		return err
 	}
-	metrics.DumpToLog(ctx)
+	// TODO(lostluck) 2020/01/24: What's the right way to expose the
+	// metrics store for the direct runner?
+	metrics.DumpToLogFromStore(ctx, plan.Store)
```

Comment:
To answer the TODO's question: This seems fine to me. What are the concerns you have about exposing the metrics like this?

Reply:
It's more of a general comment of how do we give users programmatic access to the metrics after pipeline completion. So there's nothing wrong with the way the direct runner dumps the metrics every time. That's fine.

We likely need an extractor registration set up, to go along with the runner Execute registration, as in the absence of a reliable job management server, there's no common way for runners to return metrics to the users.